ts/Dashboards/Components/DataGridComponent/DataGridComponent.ts

```diff
@@ -129,338 +137,88 @@ class DataGridComponent extends Component {
-        const { dataGrid } = this;
-        const connector = this.getFirstConnector();
-
-        if (connector && dataGrid) {
-            dataGrid.on('cellClick', (e: any): void => {
-                if ('input' in e) {
-                    e.input.addEventListener(
-                        'keyup',
-                        (keyEvent: any): void =>
-                            this.options.onUpdate(keyEvent, connector)
-                    );
-                }
-            });
-        }
-    }
-
-    /**
-     * Based on the `visibleColumns` option, filter the columns of the table.
-     *
-     * @internal
-     */
-    private filterColumns(): DataTable|undefined {
-        const table = this.getFirstConnector()?.table.modified,
-            visibleColumns = this.options.visibleColumns;
-
-        if (table) {
-            // Show all columns if no visibleColumns is provided.
-            if (!visibleColumns?.length) {
-                return table;
-            }
-
-            const columnsToDelete = table
-                .getColumnNames()
-                .filter((columnName): boolean => (
-                    visibleColumns?.length > 0 &&
-                    // Don't add columns that are not listed.
-                    !visibleColumns.includes(columnName)
-                    // Else show the other columns.
-                ));
-
-            // On a fresh table clone remove the columns that are not mapped.
-            const filteredTable = table.clone();
-            filteredTable.deleteColumns(columnsToDelete);
-
-            return filteredTable;
-        }
-    }
-
-    public getOptionsOnDrop(sidebar: SidebarPopup): Partial<Options> {
-        const connectorsIds =
-            sidebar.editMode.board.dataPool.getConnectorIds();
-        let options: Partial<Options> = {
-            cell: '',
-            type: 'DataGrid'
-        };
-
-        if (connectorsIds.length) {
-            options = {
-                ...options,
-                connector: {
-                    id: connectorsIds[0]
-                }
-            };
+    public override resize(
+        width?: number | string | null,
+        height?: number | string | null
+    ): void {
+        if (height) {
+            this.contentElement.style.minHeight = '0';
+        } else if (height === null) {
+            this.contentElement.style.removeProperty('min-height');
         }
 
-        return options;
+        super.resize(width, height);
+        this.dataGrid?.viewport?.reflow();
     }
 
-    /** @private */
-    public toJSON(): DataGridComponent.ClassJSON {
-        const dataGridOptions = JSON.stringify(this.options.dataGridOptions);
-        const base = super.toJSON();
-
-        const json = {
-            ...base,
-            options: {
-                ...base.options,
-                dataGridOptions
-            }
-        };
-
-        this.emit({ type: 'toJSON', json });
-        return json;
+    public onTableChanged(): void {
```

Comment:
Add docklet

Reply:
It's an overrided method, so it's not needed here. I'll add the `override` keyword to make it more clear.